crates/core/src/config.rs

```diff
@@ -30,7 +30,12 @@ impl Config {
     /// Initializes the config
     /// Must be called before any call to `instance`
     pub unsafe fn initialize(mut config: Self) {
-        if INSTANCE.get().is_some() { panic!("Can only initialize core config once"); }
+        Self::initialize_test_sensitive(config, false);
+    }
+
+    pub unsafe fn initialize_test_sensitive(mut config: Self, is_test: bool) {
+        // We may initialize many times in test mode
+        if INSTANCE.get().is_some() && !is_test { panic!("Can only initialize core config once"); }
```

Comment:
You could do:

```rust
#[cfg(not(test))]
pub unsafe fn initialize(mut config: Self) { ... }

#[cfg(test)]
pub unsafe fn initialize(mut config: Self) { ... }
```

This would make the test initialize different

Reply:
I still wanted to test that the config can't be initialized twice, didn't know how to do that with the `#[cfg(test)]` approach